dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -1,70 +1,175 @@
+
+    def validate_config(self) -> None:
+        """Perform any validation required to ensure submission method can proceed."""
+        pass
+
+    @abstractmethod
+    def build_submitter(self) -> PythonSubmitter:
         """
-        The workflow gets modified throughout. Settings added through dbt are popped off
-        before the spec is sent to the Databricks API
+        Since we don't own instantiation of the Helper, we construct the submitter here,
+        after validation.
         """
-        return self.parsed_model["config"].get("workflow_job_config", {})
+        pass
 
-    @property
-    def cluster_spec(self) -> Dict[str, Any]:
-        return self.parsed_model["config"].get("job_cluster_config", {})
+    def submit(self, compiled_code: str) -> None:
+        self.command_submitter.submit(compiled_code)
 
-    def __init__(self, parsed_model: Dict, credentials: DatabricksCredentials) -> None:
-        self.credentials = credentials
-        self.identifier = parsed_model["alias"]
-        self.schema = parsed_model["schema"]
-        self.database = parsed_model.get("database")
-        self.parsed_model = parsed_model
-        use_user_folder = parsed_model["config"].get("user_folder_for_python", False)
 
-        self.check_credentials()
+class PythonCommandSubmitter(PythonSubmitter):
+    """Submitter for Python models using the Command API."""
 
-        self.api_client = DatabricksApiClient.create(
-            credentials, self.get_timeout(), use_user_folder
+    def __init__(
+        self, api_client: DatabricksApiClient, tracker: PythonRunTracker, cluster_id: str
+    ) -> None:
+        self.api_client = api_client
+        self.tracker = tracker
+        self.cluster_id = cluster_id
+
+    @override
+    def submit(self, compiled_code: str) -> None:
+        context_id = self.api_client.command_contexts.create(self.cluster_id)
+        command_exec: Optional[CommandExecution] = None
+        try:
+            command_exec = self.api_client.commands.execute(
+                self.cluster_id, context_id, compiled_code
+            )
+
+            self.tracker.insert_command(command_exec)
+            # poll until job finish
+            self.api_client.commands.poll_for_completion(command_exec)
+
+        finally:
+            if command_exec:
+                self.tracker.remove_command(command_exec)
+            self.api_client.command_contexts.destroy(self.cluster_id, context_id)
+
+
+class PythonNotebookUploader:
+    """Uploads a compiled Python model as a notebook to the Databricks workspace."""
+
+    def __init__(
+        self, api_client: DatabricksApiClient, catalog: str, schema: str, identifier: str
+    ) -> None:
+        self.api_client = api_client
+        self.catalog = catalog
+        self.schema = schema
+        self.identifier = identifier
+
+    @staticmethod
+    def create(
+        api_client: DatabricksApiClient, parsed_model: ParsedPythonModel
+    ) -> "PythonNotebookUploader":
```

Comment:
why not pass `parsed_model` as param of the constructor

Reply:
this case is simple enough that you're probably right.  I was trying to move construction logic out into create methods, but this is so simple that it isn't justified, since there is no conditional logic in it.